db/migrate/20201211202144_add_color_for_predicates.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class AddColorForPredicates < ActiveRecord::Migration[6.0]
+  def change
+    change_table :predicates do |t|
+      t.string :color
+    end
+
+    ###
+    # @description: This change might affect existent records. The solution is to run the "assign_color" method
+    #   to ensure the field is correctly populated to all those that hasn't.
+    #   If, for any reason, this method doesn't remain in the future, replace it by assigning the color using
+    #   the console or a rake task.
+    ###
+    Predicate.all.each {|p|
+      p.assign_color if p.respond_to?(:assign_color) && !p.color.present?
```

Comment:
I think it requires a `save` after the assignment. Also, better replace `Predicate.all.each` with `Predicate.find_each`, so the records get loaded in batches and not all at once.

Reply:
It does require a `save!`. Changed.